crypto/ecdsa/utils.go

```diff
@@ -0,0 +1,73 @@
+package ecdsa
+
+import (
+	"bufio"
+	"crypto/ecdsa"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/google/uuid"
+)
+
+func WriteKey(path string, privateKey *ecdsa.PrivateKey, password string) error {
+	UUID, err := uuid.NewRandom()
+	if err != nil {
+		return err
+	}
+	key := &keystore.Key{
+		Id:         UUID,
+		Address:    crypto.PubkeyToAddress(privateKey.PublicKey),
+		PrivateKey: privateKey,
+	}
+
+	encryptedBytes, err := keystore.EncryptKey(key, password, keystore.StandardScryptN, keystore.StandardScryptP)
+	if err != nil {
+		return err
+	}
+
+	return writeBytesToFile(path, encryptedBytes)
+}
+
+func writeBytesToFile(path string, data []byte) error {
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		fmt.Println("Error creating directories:", err)
```

Comment:
does `os.MkdirAll` error if the directory already exists? Weird to print an error like this if so.

Reply:
no if it's already created - it is a noop. - only if it's not able to create anything it gives error.

Reply:
lol https://pkg.go.dev/os#MkdirAll says it's noop (If path is already a directory, MkdirAll does nothing and returns nil)  but not Mkdir